pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,725 @@
+    CIM_ERR_INVALID_NAMESPACE, CIM_ERR_FAILED, CIM_ERR_INVALID_PARAMETER, \
+    CIM_ERR_NOT_FOUND, CIM_ERR_ALREADY_EXISTS, \
+    CIMInstanceName, CIM_ERR_NAMESPACE_NOT_EMPTY
+
+from pywbem._nocasedict import NocaseDict
+
+from pywbem._utils import _format
+from ._resolvermixin import ResolverMixin
+
+__all__ = ['InMemoryRepository']
+
+
+class InMemoryRepository(BaseRepositoryConnection, ResolverMixin):
+    """
+    Create an adaption of the MOF compiler MOFWBEMConnection class so that to
+    directly use the attributes that represent the repository and implement a
+    modified CreateClass that resolves the new class. This directs the compiler
+    output directly  to the dictionaries for qualifiers, and instances in the
+    FakedWBEMConnection object and replaces the CreateClass with a local method
+    that allows resolving the created class before it is inserted into the
+    repository.
+
+    This class is private to pywbem_mock
+    """
+    def __init__(self, conn):
+        """
+        Parameters:
+
+          conn (BaseRepositoryConnection):
+            The underlying repository connection.
+
+            `None` means that there is no underlying repository and all
+            operations performed through this object will fail.
+        """
+
+        self.conn = conn
+        self.conn_id = conn.conn_id
+        self.class_names = {}
+        # The CIM classes in the mock repository.
+        # This is a dictionary of dictionaries where the top level key is the
+        # CIM namespace name and the keys for each sub-dictionary in a
+        # namespace are class names, and the values in each sub-dictionary are
+        # the CIM classes in that namespace, represented as CIMClass objects.
+        # The dictionaries are NocaseDict since namespaces should be case
+        # insensitive.
+        self.classes = NocaseDict()
+
+        # The CIM qualifier types in the mock repository.
+        # Same format as for classes above, except that the values in each
+        # sub-dictionary are CIMQualifierDeclaration objects.
+        self.qualifiers = NocaseDict()
+
+        # The CIM instances in the mock repository.
+        # Because instances do not have a name, the format is slightly
+        # different: This is a dictionary of lists where the top level key is
+        # the CIM namespace name and the value is a list of CIM instances in
+        # that namespace, represented as CIMInstance objects.
+        # TODO: ks. FUTURE maybe we should really have a subdict per class but
+        #  it is probably not important for initial release.
+        self.instances = {}
+
+        # Methods that will be executed. This is a dictionary of dictionaries
+        # where the top level is the CIMNamespace, the next level is the
+        # class in which the method will execute, and the third level defines
+        # the method to be executed.  The value is the callback method to
+        # be executed
```

Comment:
1. Mention the name of the interface the callback method needs to satisfy.
2. Wording: "third level" is confusing if there are only two levels of dictionaries. It is the value of the second level of dictionaries.

Reply:
outdated, withdrawing comment.